ConsentViewController/Classes/ConsentViewController.swift

```diff
@@ -407,38 +416,47 @@ import Reachability
         userDefaults.setValue(String(parsedPurposeConsents), forKey: ConsentViewController.IAB_CONSENT_PARSED_PURPOSE_CONSENTS)
     }
 
-    private func isDefined(_ string: String) -> Bool {
-        return string != "undefined"
+    private func onReceiveMessage(willShow: Bool) {
+        willShow ? onMessageReady?(self) : done()
     }
 
-    /// :nodoc:
-    public func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
-        guard
-            let messageBody = message.body as? [String: Any],
-            let name = messageBody["name"] as? String,
-            let body = messageBody["body"] as? [String: Any?]
-        else { return }
+    private func onMessageChoiceSelect(choiceType: Int) {
+        guard Reachability()!.connection != .none else {
+            onErrorOccurred?(NoInternetConnection())
+            done()
+            return
+        }
+        self.choiceType = choiceType
+        onMessageChoiceSelect?(self)
+    }
+
+    private func onInteractionComplete(euconsent: String, consentUUID: String) {
+        self.euconsent = euconsent
+        self.consentUUID = consentUUID
+        do {
+            try storeIABVars(euconsent)
+            let userDefaults = UserDefaults.standard
+            userDefaults.setValue(euconsent, forKey: ConsentViewController.EU_CONSENT_KEY)
+            userDefaults.setValue(consentUUID, forKey: ConsentViewController.CONSENT_UUID_KEY)
+            userDefaults.synchronize()
+        } catch let error as ConsentViewControllerError {
+            onErrorOccurred?(error)
+        } catch {}
+        done()
+    }
+
+    private func onErrorOccurred(errorType: String) {
+        onErrorOccurred?(WebViewErrors[errorType] ?? PrivacyManagerUnknownError())
+        done()
+    }
 
+    private func handleMessage(withName name: String, andBody body: [String:Any?]) {
         switch name {
         case "onReceiveMessageData": // when the message is first loaded
-            if(body["willShowMessage"] as! Int) == 1 {
-                webView.frame = webView.superview!.bounds
-                willShowMessage?(self)
-                return
-            }
-            done()
+            onReceiveMessage(willShow: body["willShowMessage"] as! Int == 1)
```

Comment:
Is there any risk that these values could not be `Int` (if the JSON wasn't quite right, for example?)

If it could, it would be nice to have this as an optional check, and if it can't be parsed as an `Int` call the default behaviour for the switch.

Reply:
Since I treat SourcePoint's api as a black box, we should prepare for the case when the JSON isn't well formed. So yeah, I'll make it optional checks.